est.nf

```diff
@@ -58,6 +72,11 @@ process cat_fasta_files {
     """
     cat $input > all_sequences.fasta
     """
+    if (params.import_mode == "blast") {
+        """
+        perl $projectDir/src/est/import/append_blast_query.pl --blast-query-file ${params.blast_query_file} --output-sequence-file all_sequences.fasta
+        """
+    }
```

Comment:
This won't work the way I think you intended it to. A nextflow process [is a Groovy function that returns a string](https://www.nextflow.io/docs/latest/process.html#conditional-scripts). The last thing on the stack here could be nothing if the condition evaluates to false. You need to build a string here instead.
```
    cat_cmd = "cat $input > all_sequences.fasta"
    if (params.import_mode == "blast") {
        """
        $cat_cmd
        perl $projectDir/src/est/import/append_blast_query.pl --blast-query-file ${params.blast_query_file} --output-sequence-file all_sequences.fasta
        """
    } else {
        cat_cmd
    }
```